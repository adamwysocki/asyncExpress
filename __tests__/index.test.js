const request = require("supertest");
const app = require("../dist/index.js");
const _ = require("lodash");

describe("index route", () => {
  expect.extend({
    toLookLike(received, expected) {
      received = JSON.parse(received);
      const pass = _.isEqual(received, expected);

      const message = pass
        ? () =>
            `${this.utils.matcherHint(".not.toLookLike")}
                \n\n
                Expected value to not be (using Object.is):\n
                  ${this.utils.printExpected(expected)}\n
                Received:\n
                  ${this.utils.printReceived(received)}`
        : () => {
            const diffString = diff(expected, received, {
              expand: this.expand
            });
            return `${this.utils.matcherHint(".toLookLike")} 
                  \n\n
                  Expected value to be (using Object.is):
                    ${this.utils.printExpected(expected)}
                    Received:
                    ${this.utils.printReceived(received)}
                  ${diffString} ? \n\nDifference:\n\n${diffString}}`;
          };

      return { actual: received, message, pass };
    }
  });

  afterAll(() => {
    app.server.close();
  });

  it("should respond with a 200", () => {
    return request(app)
      .get("/")
      .expect(200)
      .expect(response => {
        expect(response.text).toMatch(/title/);
        expect(response.text).toMatch(/Google/);
      });
  });

  it("should respond with a title", () => {
    return request(app)
      .get("/")
      .expect(response => {
        expect(response.text).toMatch(/title/);
      });
  });

  it("should respond with the correct JSON data", () => {
    return request(app)
      .get("/")
      .expect(response => {
        expect(response.text).toLookLike({ result: "success", code: 200, data: { title: "Google" } });
      });
  });

  it("should render correctly", () => {
    return request(app)
      .get("/")
      .expect(response => {
        expect(response.text).toMatchSnapshot();
      });
  });
});
