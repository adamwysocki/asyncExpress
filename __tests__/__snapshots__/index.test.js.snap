// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index route should render correctly 1`] = `"{\\"result\\":\\"success\\",\\"code\\":200,\\"data\\":{\\"title\\":\\"CaptainU\\"}}"`;
